src/test/java/org/candlepin/resource/OwnerContentResourceTest.java

```diff
@@ -0,0 +1,297 @@
+
+        assertNotNull(response);
+
+        Collection<ContentDTO> received = response.list();
+
+        assertEquals(0, received.size());
+    }
+
+    @Test
+    public void getOwnerContent() {
+        Owner owner = this.createOwner("test_owner");
+        Content content = this.createContent("test_content", "test_content", owner);
+        ContentDTO output = this.ownerContentResource.getOwnerContent(owner.getKey(), content.getId());
+
+        assertNotNull(output);
+        assertEquals(content.getId(), output.getId());
+    }
+
+    @Test
+    public void getOwnerContentNotFound() {
+        Owner owner = this.createOwner("test_owner");
+
+        assertThrows(NotFoundException.class, () ->
+            this.ownerContentResource.getOwnerContent(owner.getKey(), "test_content")
+        );
+    }
+
+    @Test
+    public void createContent() {
+        Owner owner = this.createOwner("test_owner");
+        ContentDTO cdto = TestUtil.createContentDTO("test_content");
+        cdto.setLabel("test-label");
+        cdto.setType("test-test");
+        cdto.setVendor("test-vendor");
+
+        assertNull(this.ownerContentCurator.getContentById(owner, cdto.getId()));
+
+        ContentDTO output = this.ownerContentResource.createContent(owner.getKey(), cdto);
+
+        assertNotNull(output);
+        assertEquals(cdto.getId(), output.getId());
+
+        Content entity = this.ownerContentCurator.getContentById(owner, cdto.getId());
+        assertNotNull(entity);
+        assertEquals(cdto.getName(), entity.getName());
+        assertEquals(cdto.getLabel(), entity.getLabel());
+        assertEquals(cdto.getType(), entity.getType());
+        assertEquals(cdto.getVendor(), entity.getVendor());
+    }
+
+    @Test
+    public void createContentWhenContentAlreadyExists()  {
+        // Note:
+        // The current behavior of createContent is to update content if content already exists
+        // with the given RHID. So, our expected behavior in this test is to trigger an update.
+
+        Owner owner = this.createOwner("test_owner");
+        Content content = this.createContent("test_content", "test_content", owner);
+        ContentDTO cdto = TestUtil.createContentDTO("test_content", "updated_name");
+        cdto.setLabel("test-label");
+        cdto.setType("test-test");
+        cdto.setVendor("test-vendor");
+
+        assertNotNull(this.ownerContentCurator.getContentById(owner, cdto.getId()));
+
+        ContentDTO output = this.ownerContentResource.createContent(owner.getKey(), cdto);
+
+        assertNotNull(output);
+        assertEquals(cdto.getId(), output.getId());
+        assertEquals(cdto.getName(), output.getName());
+
+        Content entity = this.ownerContentCurator.getContentById(owner, cdto.getId());
+        assertNotNull(entity);
+        assertEquals(cdto.getName(), entity.getName());
+    }
+
+    @Test
+    public void createContentWhenContentAlreadyExistsAndLocked()  {
+        // Note:
```

Comment:
This note is no longer in line with the actual behavior of the test, so let's remove it.